app/models/blog.rb

```diff
@@ -10,7 +10,7 @@ class Blog < ApplicationRecord
   scope :published, -> { where('secret = FALSE') }
 
   scope :search, lambda { |term|
-    where("title LIKE '%#{term}%' OR content LIKE '%#{term}%'")
+    where('title LIKE ? OR content LIKE ?', "%#{sanitize_sql_like(term)}%", "%#{sanitize_sql_like(term)}%") if term.present?
```

Comment:
```suggestion
    where('title LIKE :hoge OR content LIKE :hoge', ...
```

のように名前付きのパラメータにすると少しコードがDRYになります。
あと、`if term.present?`は付けなくて結果として問題なく動作すると思うんですが、いかがでしょう？

Reply:
<img src="https://github.com/ckym02/haunted-blog/assets/50766994/8a2843cc-536e-4ad7-988a-8fd76abf0f15" alt="" width=600 height=500>